feat: Added approve method for Mergerequests

Joustie opened the pull request:
Offical GitLab API supports approval for GitLab EE

gpocentek commented on gitlab/v4/objects.py:
```diff
@@ -2132,6 +2132,30 @@ def changes(self, **kwargs):
+    @cli.register_custom_action('ProjectMergeRequest', tuple(), ('sha'))
+    @exc.on_http_error(exc.GitlabMRApprovalError)
+    def approve(self,
+                sha,
```
`sha` is optional so it would be best to have a default value.
```suggestion
                sha=None,
```

Could you also have all the parameters on one line (purely cosmetic)?

Joustie replied:
Sure no problem.

Joustie commented:
I have added the unapprove method and I have put the parameters on one line, added default value for sha. I could not find the pep8 errors about double missing blank lines? (I ran tox locally against my branch).

max-wittig commented:
@gpocentek Travis is not reporting back and I don't see the build in the UI. Do you have any way to restart it?

Joustie commented:
I am not very familiar with travis, I have closed and reopened the pull request. That should trigger another build?

max-wittig commented:
A force push would trigger another build. But I think @gpocentek should be able to restart it manually.

gpocentek commented:
I can't see the build on travis.

@Joustie Do you mind doing a rebase and a push --force? Thanks!

Joustie commented:
@gpocentek I have made sure the travis builds succeed for my fork, and rebased and pushed --force but still the checks are not updated? Maybe try a brandnew PR?

max-wittig commented:
@Joustie I'm sorry that travis is misbehaving. I think we should merge it. Tests pass locally for me.

@Joustie Could you do a rebase and then @gpocentek could merge it.

gpocentek commented:
@Joustie the changes look good and we'll merge without travis, but there are merge conflicts that need to be resolved first. Could you have a look at that?

Thanks!
